src/Datadog.Trace/Propagation/CompositePropagatorsProvider.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Datadog.Trace.Conventions;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Plugins;
+
+namespace Datadog.Trace.Propagation
+{
+    internal class CompositePropagatorsProvider
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<CompositePropagatorsProvider>();
+
+        private readonly ICollection<IPropagatorsProvider> _providers;
+
+        public CompositePropagatorsProvider()
+        {
+            _providers = new List<IPropagatorsProvider>();
+        }
+
+        public CompositePropagatorsProvider RegisterProvider(IPropagatorsProvider provider)
+        {
+            _providers.Add(provider);
+
+            return this;
+        }
+
+        public CompositePropagatorsProvider RegisterProviderFromPlugins(IEnumerable<IOTelPlugin> plugins)
+        {
+            foreach (var plugin in plugins)
+            {
+                _providers.Add(plugin.GetPropagatorsProvider());
```

Comment:
I'm not sure if I'm overengineering here, but if every plugin needs to have a  `GetPropagatorsProvider()` and a plugin may not be affecting propagators, this could return `null`. Would it make sense to add a `HasPropagatorsProvider` to the `IOTelPlugin` interface right now?

Reply:
@zacharycmontoya We need to have a single small interface per extension point. Why? To ensure future compatibility. We do not want to require rebuilding and shipping a new plugin if OTel .NET AutoInstr provides a new extension point in the same interface.

@RassK In general it is good to describe to write some docs as part of PR. It not only helps to maintain the docs up to date but also helps a lot during the review process. Especially that the PR's description does not tell much 😉

Reply:
Had to completely refactor this. At first it was quite hard to imagine exact use case of multiple extension points, but I think we are getting closer.